sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -57,191 +55,106 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * A {@link CoderRegistry} allows registering the default {@link Coder} to use for a Java class,
- * and looking up and instantiating the default {@link Coder} for a Java type.
+ * A {@link CoderRegistry} allows creating a {@link Coder} for a given Java {@link Class class} or
+ * {@link TypeDescriptor type descriptor}.
  *
- * <p>{@link CoderRegistry} uses the following mechanisms to determine a default {@link Coder} for a
- * Java class, in order of precedence:
- * <ol>
- *   <li>Registration:
- *     <ul>
- *       <li>A {@link CoderFactory} can be registered to handle a particular class via
- *           {@link #registerCoder(Class, CoderFactory)}.</li>
- *       <li>A {@link Coder} class with the static methods to satisfy
- *           {@link CoderFactories#fromStaticMethods} can be registered via
- *           {@link #registerCoder(Class, Class)}.</li>
- *       <li>Types can be automatically registered via {@link CoderRegistrar coder registrars}.</li>
- *     </ul>
- *   <li>Annotations: {@link DefaultCoder} can be used to annotate a type with
- *       the default {@code Coder} type. The {@link Coder} class must satisfy the requirements
- *       of {@link CoderProviders#fromStaticMethods}.
- *   <li>Fallback: A fallback {@link CoderProvider} is used to attempt to provide a {@link Coder}
- *       for any type. By default, there is {@link SerializableCoder#PROVIDER}, which can provide a
- *       {@link Coder} for any type that is serializable via Java serialization. The fallback
- *       {@link CoderProvider} can be get and set respectively using
- *       {@link #getFallbackCoderProvider()} and {@link #setFallbackCoderProvider}. Multiple
- *       fallbacks can be chained together using {@link CoderProviders#firstOf}.
- * </ol>
+ * <p>Creation of the {@link Coder} is delegated to one of the many registered
+ * {@link CoderFactory coder factories} based upon the registration order.
+ *
+ * <p>By default, the {@link CoderFactory coder factory} registration order is as follows:
+ * <ul>
+ *   <li>Coder factories registered programmatically with
+ *       {@link CoderRegistry#registerCoderFactory(CoderFactory)}.</li>
```

Comment:
Naming fun: shorten to `register` and all the overloads can be the same, though you could have `registerCoderFactory` and `registerCoder` separate, too.